code/modules/atmospherics/machinery/datum_pipeline.dm

```diff
@@ -90,11 +90,10 @@
 			merge(P.parent)
 		P.parent = src
 		var/list/adjacent = P.pipeline_expansion()
-		for(var/obj/machinery/atmospherics/pipe/I in adjacent)
-			if(I.parent == src)
-				continue
-			var/datum/pipeline/E = I.parent
-			merge(E)
+		for(var/obj/machinery/atmospherics/pipe/I in adjacent - members)
+			//if(I.parent == src)
+			//	continue
+			merge(I.parent)
```

Comment:
Why did you change this?

Reply:
i think about remove all checks like  `f(I.parent == src)` 
i think it redundant

Reply:
I don't think it is. The check should be there to prevent adding mols from yourself into yourself.

As far as I can tell you're attempting to fix that with the subtraction of members from adjacent, but that appears to be built to prevent merging with itself.

So it does what your subtraction does but without having to modify the list itself. I'm not sure if this is a performance benefit or downside, but you would need to profile it separately from all your changes here.